web/vue.config.js

```diff
@@ -1,6 +1,7 @@
 const MomentLocalesPlugin = require('moment-locales-webpack-plugin');
+const child_process = require('child_process');
 
-process.env.VUE_APP_VERSION = process.env.COMMIT_REF;
+process.env.VUE_APP_VERSION = child_process.execSync("git describe").toString();
```

Comment:
i think for dandiarchive is fine, but this would mean any current or future deployment would need to have git installed locally and that we are cloning rather than say downloading a zip file and building. 

@yarikoptic - as an example, in heroku for api there is no git clone (not even a possibility). we looked into it there. the standard django deployment simply downloads the latest zip file.

Reply:
This is a good point.

The fix is to build in error detection and fall back to the git sha. I will work on this in a bit.

Reply:
As far as dandi-api is concerned, there my objections to attaching a semver don't really exist, and if we want to buy into it, something like semantic-release could be very useful, and would obviate the need to scrounge up git shas to display in the API.

Reply:
FWIW -- I think `git describe`, which does include git shas past the tag/release point, would always be useful as long as there is a chance to get a "devel" version running. Yes -- comes at a cost of having/needing git to be installed.  
If we ever to provide a proper "distribution" of the dandiarchive etc, then static version string could be minted upon generation of the distirbution.  That is what e.g. `versionner` does for source tarballs for distribution/upload to pypi, making it not needed to have `git` upon installation for that "static-ly installed" version.

Reply:
> The fix is to build in error detection and fall back to the git sha. I will work on this in a bit.

This is done now (see ac94847).

BTW, this does not completely solve the problem of how to obtain a version number for the case when this is being built neither locally, nor via Netlify. I think that would require a bigger discussion and a design doc.

But for now I believe this is ready for another review.